app/uk/gov/hmrc/gform/validation/ValidationUtil.scala

```diff
@@ -165,10 +168,11 @@ object ValidationUtil {
         val linesMap = List("line1", "line2", "line3").zip(lines).toMap
         val otherMap = Map(
           "city"     -> city.headOption.getOrElse(""),
-          "postcode" -> itmpAddress.flatMap(_.postCode).getOrElse(""),
-          "country"  -> itmpAddress.flatMap(_.countryName).getOrElse("")
+          "postcode" -> itmpAddress.flatMap(_.postCode).getOrElse("")
         )
-        val atomMap: Map[String, String] = linesMap ++ otherMap
+        val countryMap =
+          itmpAddress.flatMap(_.countryName).filter(_.trim.nonEmpty).map(c => Map("country" -> c)).getOrElse(Map.empty)
```

Comment:
The same here, this should not be its own map. (Or perhaps I'm missing the reason for this and if so let's put comment into the code to explain it)

Reply:
yes, it's no longer needed. I put "country" back to the original map.